src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -461,18 +453,36 @@ trait WorkflowComponent {
       )
     }
 
-    private def unmarshalInputResolution(validationRec: SubmissionValidationRecord, value: Option[SubmissionAttributeRecord]): SubmissionValidationValue = {
-      value match {
-        case Some(attr) => SubmissionValidationValue(Some(submissionAttributeQuery.unmarshalAttributes(Seq(((attr.id, attr), None)))(attr.id).values.head), validationRec.errorText, validationRec.inputName)
-        case None => SubmissionValidationValue(None, validationRec.errorText, validationRec.inputName)
-      }
+    //Unmarshal all input resolutions for a single workflow. Assumes that everything in wfInputResolutionRecs has the same workflowId.
+    def unmarshalOneWorkflowInputs(wfInputResolutionRecs: Seq[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])], workflowId: Long): Seq[SubmissionValidationValue] = {
+
+        //collect up the workflow resolution results by input
+        val resolutionsByInput = wfInputResolutionRecs.groupBy { case (resolution, attribute) => resolution.inputName }
+
+        //unmarshalAttributes will unmarshal multiple workflow attributes at once, but it expects all the attribute records to be real and not options.
+        //To get around this, we split by input, so that each input is successful (or not) individually.
+        val submissionValues = resolutionsByInput map { case (inputName, recTuples: Seq[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])]) =>
+          val attr = if( recTuples.forall { case (submissionRec, attrRecOpt) => attrRecOpt.isDefined } ) {
+            //all attributes are real
+            Some(
+              submissionAttributeQuery.unmarshalAttributes( recTuples map { case (rec, attrOpt) => ((workflowId, attrOpt.get), None) } )(workflowId)(inputName)
+            )
+          } else {
+            None
+          }
+          //assuming that the first elem has the error here
+          SubmissionValidationValue(attr, recTuples.head._1.errorText, inputName)
+        }
+        submissionValues.toSeq
     }
 
-    private def unmarshalInputResolutions(resolutions: WorkflowQueryWithInputResolutions) = {
-      resolutions.result map { inputResolutionRecord =>
-        inputResolutionRecord map { case (inputResolution, attribute) =>
-          unmarshalInputResolution(inputResolution, Option(attribute))
```

Comment:
here's the bug (again!). iterating over each resolution record gives you each of the members in turn, so you get a list of each array element in turn, all with the same input resolution name, and then when you turn it into a map, it stomps on the values! FUN TIMES.

the lesson is this: `submissionAttributeQuery.unmarshalAttributes` is the function you want, always. just give it all of the attributes, regardless of who owns them. it will sort it out, and it will do a far better job than you